### ISSUE TITLE
Dynamic registration of apps does not allow the setting of re-write headers.

### ISSUE THREAD

**Issue by srelf-eschercloud** [gravitational/teleport#20189]
Expected behavior:
When using the dynamic registration of an application, setting of headers/re-write rules works

Current behavior:
Having a yaml file with rewrite headers set throws an error.

Bug details:
- Teleport version 11.1.2
- Recreation steps

create yaml file like the below

```yaml
kind: app
version: v3
metadata:
  name: server-ilo
  description: ""
  labels:
    env: prod
    service: ilo
spec:
  uri: https://server-ilo.example.com/
  public_addr:server-ilo.example.com.teleport.example.com
  insecure_skip_verify: true
  rewrite:
    headers:
    - "Connection: Upgrade"
    - "Upgrade: websocket"
```

Run  tctl create server.yaml
The following error is thrown:

```
ERROR: types.AppV3.Spec: types.AppSpecV3.Rewrite: types.Rewrite.Headers: []*types.Header: readObjectStart: expect { or n, but found ", error found in #10 byte of ...|eaders":["Connection|..., bigger context ...|eport...com","rewrite":{"headers":["Connection: Upgrade","Upgrade: websocket"]},"uri":|...

adjusting 
  rewrite:
    headers:
    - {Connection: Upgrade}
    - {Upgrade: websocket}
```

AS its looking for a "{" completes, but does not set the headers.

- Debug logs

**Comment by swastik959** [comment]
hi I want to work on this issue can you guide me through it

**Comment by webvictim** [comment]
I **think** that this is the same root issue that https://github.com/gravitational/teleport/pull/35222 fixes the docs for.

Try this instead:

```yaml
spec:
  rewrite:
    headers:
    - name: "Connection"
      value: "Upgrade"
    - name: "Upgrade"
      value: "websocket"
```

**Comment by zmb3** [comment]
Yes, this has always worked and the docs were just wrong, which was fixed in #35222
